src/main/java/org/opensearch/ad/transport/ADTaskProfileTransportAction.java

```diff
@@ -12,75 +12,83 @@
-        TransportService transportService,
+        SDKClusterService sdkClusterService,
         ActionFilters actionFilters,
         ADTaskManager adTaskManager,
-        HashRing hashRing
+        /* MultiNode support https://github.com/opensearch-project/opensearch-sdk-java/issues/200 */
+        // HashRing hashRing,
+        TaskManager taskManager
     ) {
-        super(
-            ADTaskProfileAction.NAME,
-            threadPool,
-            clusterService,
-            transportService,
-            actionFilters,
-            ADTaskProfileRequest::new,
-            ADTaskProfileNodeRequest::new,
-            ThreadPool.Names.MANAGEMENT,
-            ADTaskProfileNodeResponse.class
-        );
+        super(ADTaskProfileAction.NAME, actionFilters, taskManager);
         this.adTaskManager = adTaskManager;
-        this.hashRing = hashRing;
+        /* MultiNode support https://github.com/opensearch-project/opensearch-sdk-java/issues/200 */
+        // this.hashRing = hashRing;
+        this.sdkClusterService = sdkClusterService;
     }
 
-    @Override
     protected ADTaskProfileResponse newResponse(
         ADTaskProfileRequest request,
         List<ADTaskProfileNodeResponse> responses,
         List<FailedNodeException> failures
     ) {
-        return new ADTaskProfileResponse(clusterService.getClusterName(), responses, failures);
+        return new ADTaskProfileResponse(sdkClusterService.state().getClusterName(), responses, failures);
     }
 
-    @Override
     protected ADTaskProfileNodeRequest newNodeRequest(ADTaskProfileRequest request) {
         return new ADTaskProfileNodeRequest(request);
     }
 
-    @Override
     protected ADTaskProfileNodeResponse newNodeResponse(StreamInput in) throws IOException {
         return new ADTaskProfileNodeResponse(in);
     }
 
     @Override
-    protected ADTaskProfileNodeResponse nodeOperation(ADTaskProfileNodeRequest request) {
+    protected void doExecute(Task task, ADTaskProfileRequest request, ActionListener<ADTaskProfileResponse> actionListener) {
+        /* @anomaly.detection Commented until we have extension support for hashring : https://github.com/opensearch-project/opensearch-sdk-java/issues/200
         String remoteNodeId = request.getParentTask().getNodeId();
         Version remoteAdVersion = hashRing.getAdVersion(remoteNodeId);
+         */
+        Version remoteAdVersion = Version.CURRENT;
         ADTaskProfile adTaskProfile = adTaskManager.getLocalADTaskProfilesByDetectorId(request.getDetectorId());
-        return new ADTaskProfileNodeResponse(clusterService.localNode(), adTaskProfile, remoteAdVersion);
+        actionListener
+            .onResponse(
+                newResponse(
+                    request,
+                    new ArrayList<>(List.of(new ADTaskProfileNodeResponse(sdkClusterService.localNode(), adTaskProfile, remoteAdVersion))),
+                    new ArrayList<>()
+                )
+            );
```

Comment:
I see you added these new changes and were not there before because all of the APIs of profile detectors were not tested. Let's test all the APIs from next time.

Reply:
Yes, I acknowledge that.